pkg/packageinstall/packageinstall.go

```diff
@@ -248,12 +241,14 @@ func (pi *PackageInstallCR) kcVersionConstraintsSatisfied(pkg *datapkgingv1alpha
 		return true
 	}
 
-	v, err := dropVersionPreReleaseAndBuild(pi.controllerVersion)
+	v, err := pi.compInfo.KappControllerVersion()
 	if err != nil {
-		pi.log.Error(err, "Unable to parse kapp-controller version", "version string", pi.controllerVersion)
 		return false
 	}
 
+	v.Pre = semver.PRVersion{}
+	v.Build = semver.BuildMeta{}
```

Comment:
should this move into ComponentInfo? seems its more related to version parsing rather than package matching

Reply:
The constraint matching is specific to packageinstall, so I think it makes more sense here. The kapp-controller version should always be as it is in the cluster, regardless of how / why we manipulate it in kapp-controller.

Reply:
I understand the point that it's not entirely semantically in-scope for the reader of PackageInstall to know about the effects of resetting the members of the version, `v` struct.   However it's less obvious to me where to move that logic, and i think it'll be a similar number of lines of code,  just maybe a little faster to read packageinstall.go until your concern is this logic specifically.